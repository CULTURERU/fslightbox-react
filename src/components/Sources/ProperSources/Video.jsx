import React from 'react';
import PropTypes from "prop-types";

const Video = (
    {
        fsLightbox: {
            elements: { sources },
            sourcesData: { videosPosters },
            data: { urls },
            core: { properSourceController }
        },
        index
    }
) => {
    const onLoadedMetaData = ({ target }) => {
        properSourceController.setIndex(index);
        properSourceController.setSourceWidth(target.videoWidth);
        properSourceController.setSourceHeight(target.videoHeight);
        properSourceController.handleLoad();
    };

    return (
        <video
            onLoadedMetadata={ onLoadedMetaData }
            className="fslightbox-single-source fslightbox-video fslightbox-opacity-0"
            controls
            ref={ sources[index] }
            poster={ videosPosters[index] }>
            <source src={ urls[index] }/>
        </video>
    );
};

Video.propTypes = {
    fsLightbox: PropTypes.object.isRequired,
    index: PropTypes.number.isRequired
};

export default Video;