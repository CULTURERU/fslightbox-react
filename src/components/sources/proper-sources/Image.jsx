import React from 'react';
import PropTypes from "prop-types";

const Image = (
    {
        fsLightbox: {
            data: { urls },
            elements: { sources },
            collections: { sourcesLoadHandlers }
        },
        index
    }
) => {
    return (
        <img
            onLoad={ sourcesLoadHandlers[index].handleLoad }
            className={ "fslightbox-source fslightbox-opacity-0" }
            ref={ sources[index] }
            src={ urls[index] }
            alt={ urls[index] }
        />
    );
};

Image.propTypes = {
    fsLightbox: PropTypes.object.isRequired,
    index: PropTypes.number.isRequired,
};

export default Image;
