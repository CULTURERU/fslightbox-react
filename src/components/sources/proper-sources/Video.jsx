import React from 'react';
import PropTypes from "prop-types";

const Video = (
    {
        fsLightbox: {
            data: { urls },
            elements: { sources },
            sourcesData: { videosPosters },
            collections: { sourcesLoadHandlers }
        },
        index
    }
) => {
    return (
        <video
            onLoadedMetadata={ sourcesLoadHandlers[index].handleLoad }
            className="fslightbox-source fslightbox-video fslightbox-opacity-0"
            controls
            ref={ sources[index] }
            poster={ videosPosters[index] }>
            <source src={ urls[index] }/>
        </video>
    );
};

Video.propTypes = {
    fsLightbox: PropTypes.object.isRequired,
    index: PropTypes.number.isRequired
};

export default Video;
