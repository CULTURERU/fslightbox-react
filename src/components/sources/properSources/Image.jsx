import React, { Component } from 'react';
import PropTypes from "prop-types";

class Image extends Component {

    constructor(props) {
        super(props);
        this.imageOnLoad = this.imageOnLoad.bind(this);
    }

    imageOnLoad(e) {
        if (this.props._.sourcesData.isSourceAlreadyLoadedArray[this.props.i]) {
            return;
        }

        this.props._.sourcesData.sourcesDimensions[this.props.i] = {
            width: e.target.width,
            height: e.target.height
        };
        this.props.onFirstSourceLoad();
    }

    render() {
        return (
            <>
                <img
                    onLoad={ this.imageOnLoad }
                    className={ "fslightbox-single-source" }
                    ref={ this.props._.elements.sources[this.props.i] }
                    src={ this.props._.data.urls[this.props.i] }
                    alt={ this.props._.data.urls[this.props.i] }
                />
            </>
        );
    }
}


Image.propTypes = {
    _: PropTypes.object.isRequired,
    i: PropTypes.number.isRequired,
    onFirstSourceLoad: PropTypes.func.isRequired,
};
export default Image;