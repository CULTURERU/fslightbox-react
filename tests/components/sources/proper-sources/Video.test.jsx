import React from 'react';
import { mount, shallow } from "enzyme";
import Video from "../../../../src/components/sources/proper-sources/Video";
import Image from "../../../../src/components/sources/proper-sources/Image";

let fsLightbox = {
    data: {
        urls: [],
    },
    sourcesData: {
        videosPosters: ["test-poster"]
    },
    elements: {
        sources: [{
            current: null
        }]
    },
    collections: {
        sourcesLoadHandlers: [{
            handleLoad: () => {}
        }]
    }
};

let video;

describe('ref to sources array in fsLightbox object', () => {
    beforeAll(() => {
        video = mount(<Video
            fsLightbox={ fsLightbox }
            index={ 0 }
        />);
    });

    it('should be equal to image', () => {
        expect(fsLightbox.elements.sources[0].current).toEqual(video.getDOMNode());
    });
});

describe('on load', () => {
    beforeAll(() => {
        fsLightbox.collections.sourcesLoadHandlers[2] = {
            handleLoad: jest.fn()
        };
        video = shallow(<Image
            fsLightbox={ fsLightbox }
            index={ 2 }
        />);
        video.simulate('load', {
            key: 'video-load-event'
        });
    });

    it('should call handleLoad', () => {
        expect(fsLightbox.collections.sourcesLoadHandlers[2].handleLoad).toBeCalledWith({
            key: 'video-load-event'
        });
    });
});

describe('Video DOM', () => {
    beforeAll(() => {
        video = shallow(<Video fsLightbox={ fsLightbox } index={ 0 }/>);
    });

    it('should match snapshot', () => {
        expect(video).toMatchSnapshot();
    });
});

