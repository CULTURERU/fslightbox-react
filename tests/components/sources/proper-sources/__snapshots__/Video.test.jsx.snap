// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video DOM should match snapshot 1`] = `
<video
  className="fslightbox-source fslightbox-video fslightbox-opacity-0"
  controls={true}
  onLoadedMetadata={[Function]}
  poster="test-poster"
>
  <source />
</video>
`;
